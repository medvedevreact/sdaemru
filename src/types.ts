export type AppartmentItem = {
  id?: string;
  title: string;
  description: string;
  price_per_day: number;
  location: string;
  metro: string;
  district: string;
  rooms: string;
  photo: string[];
  owner: {
    name: string;
    phone: string;
    email: string;
  };
};

export type HouseItem = {
  id?: string;
  title: string;
  description: string;
  price_per_day: number;
  location: string;
  metro: string;
  district: string;
  size: string;
  photo: string[];
  owner: {
    name: string;
    phone: string;
    email: string;
  };
};

export type AutoItem = {
  id?: string;
  title: string;
  description: string;
  price_per_day: number;
  location: string;
  metro: string;
  district: string;
  type: string;
  photo: string[];
  owner: {
    name: string;
    phone: string;
    email: string;
  };
};

export type AvailableType = AutoItem[] | AppartmentItem[] | HouseItem[];
