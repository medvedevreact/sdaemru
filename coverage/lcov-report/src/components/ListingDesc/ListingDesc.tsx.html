
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/ListingDesc/ListingDesc.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> / <a href="index.html">src/components/ListingDesc</a> ListingDesc.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>5/5</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>2/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>4/4</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { FaHeart } from "react-icons/fa";
import styles from "./ListingDesc.module.scss";
import { AppartmentItem, AutoItem, HouseItem } from "../../types";
&nbsp;
interface ListingDescType {
  listing: AutoItem | HouseItem | AppartmentItem;
}
&nbsp;
export const ListingDesc: React.FC&lt;ListingDescType&gt; = ({ listing }) =&gt; {
  return (
    &lt;div className={styles.wrapper}&gt;
      &lt;div className={styles.descUpper} role="desc"&gt;
        &lt;h2&gt;{listing.title}&lt;/h2&gt;
        &lt;div&gt;
          &lt;h2 className={styles.price}&gt;{listing.price_per_day} Р&lt;/h2&gt;
          &lt;p&gt;за сутки&lt;/p&gt;
        &lt;/div&gt;
      &lt;/div&gt;
      &lt;p className={styles.desc}&gt;{listing.description}&lt;/p&gt;
      &lt;div className={styles.descUnder}&gt;
        &lt;div className={styles.descUnderLeft}&gt;
          &lt;div className={styles.metro}&gt;
            &lt;svg
              fill="#664ef9"
              width="20px"
              height="20px"
              viewBox="0 0 24 24"
              role="img"
              xmlns="http://www.w3.org/2000/svg"
            &gt;
              &lt;title&gt;Moscow Metro icon&lt;/title&gt;
              &lt;path d="M16.603 11.85l-2.481-6.26-2.092 3.66-2.092-3.66-2.481 6.262H6.74v.941h3.736v-.941h-.553l.538-1.555 1.569 2.57 1.569-2.57.538 1.555h-.553v.941h3.751v-.941zm5.335-1.912A9.933 9.933 0 0 0 12 0C6.516 0 2.062 4.453 2.062 9.938c0 2.75 1.121 5.23 2.914 7.023a.804.804 0 0 0 1.375-.568.825.825 0 0 0-.239-.582 8.303 8.303 0 0 1-2.42-5.873c0-4.588 3.72-8.324 8.308-8.324 4.588 0 8.324 3.736 8.324 8.324a8.289 8.289 0 0 1-2.436 5.888l-7.024 7.023L12 24l7.039-7.039a9.891 9.891 0 0 0 2.899-7.023Z" /&gt;
            &lt;/svg&gt;
            &lt;p className={styles.metroText}&gt;{listing.metro}&lt;/p&gt;
          &lt;/div&gt;
&nbsp;
          {listing.district &amp;&amp; (
            &lt;div className={styles.metro}&gt;
              &lt;svg
                xmlns="http://www.w3.org/2000/svg"
                fill="#664ef9"
                width="20px"
                height="20px"
                viewBox="0 0 24 24"
              &gt;
                &lt;path d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z" /&gt;
              &lt;/svg&gt;
              &lt;p className={styles.metroText}&gt;{listing?.district}&lt;/p&gt;
            &lt;/div&gt;
          )}
        &lt;/div&gt;
        &lt;button className={styles.fav}&gt;
          &lt;FaHeart /&gt;В избранное
        &lt;/button&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  );
};
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-10-09T17:15:04.409Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    