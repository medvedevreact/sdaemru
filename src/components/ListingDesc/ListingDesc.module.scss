.wrapper {
  background: white;
  width: 100%;
  height: 200px;
  margin-top: 25px;
  border-radius: 15px;
  padding: 20px;
  @media (max-width: 860px) {
    padding: 15px;
    height: 175px;
  }
  @media (max-width: 600px) {
    padding: 10px;
    height: 150px;
  }
  @media (max-width: 550px) {
    padding: 10px;
    height: 160px;
    width: 100%;
  }
}
.descUpper {
  display: flex;
  justify-content: space-between;
  h2 {
    @media (max-width: 860px) {
      font-size: 18px;
      max-width: 300px;
      margin-bottom: 7px;
    }
    @media (max-width: 600px) {
      font-size: 14px;
      max-width: 200px;
      margin-bottom: 3px;
      margin-right: 5px;
    }
  }
}
.price {
  color: #664ef9;
  @media (max-width: 860px) {
    font-size: 20px;
  }
}
.desc {
  max-width: 500px;
  margin-bottom: 30px;
  @media (max-width: 880px) {
    max-width: 300px;
    font-size: 13px;
    margin-bottom: 15px;
  }
  @media (max-width: 860px) {
    margin-bottom: 10px;
  }
  @media (max-width: 700px) {
    font-size: 12px;
  }
  @media (max-width: 600px) {
    font-size: 10px;
    max-width: 250px;
  }
}
.descUnder {
  display: flex;
  justify-content: space-between;
  &Left {
    display: flex;
    gap: 20px;
  }
  .fav {
    display: flex;
    align-items: center;
    padding: 7px;
    border-radius: 12px;
    border: none;
    gap: 3px;
    color: rgb(255, 30, 30);
    cursor: pointer;
    transition: all 0.3s ease;
    &:hover {
      background: rgb(212, 212, 212);
    }
  }
}
.metro {
  display: flex;
  align-items: center;
  gap: 3px;

  &Text {
    opacity: 0.7;
    @media (max-width: 860px) {
      font-size: 13px;
    }
  }
}
.pageWrapper {
  display: flex;
  gap: 25px;
}
.leaflet-control-zoom-in,
.leaflet-control-zoom-out {
  width: 12px;
  height: 12px;
  line-height: 20px;
}

.leaflet-control-zoom-in .leaflet-icon,
.leaflet-control-zoom-out .leaflet-icon {
  font-size: 15px;
}
.leaflet-touch .leaflet-bar a {
  width: 12px !important;
  height: 12px !important;
}
