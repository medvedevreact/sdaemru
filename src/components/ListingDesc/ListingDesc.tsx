import { FaHeart } from "react-icons/fa";
import styles from "./ListingDesc.module.scss";
import { AppartmentItem, AutoItem, HouseItem } from "../../types";

interface ListingDescType {
  listing: AutoItem | HouseItem | AppartmentItem;
}

export const ListingDesc: React.FC<ListingDescType> = ({ listing }) => {
  return (
    <div className={styles.wrapper}>
      <div className={styles.descUpper}>
        <h2>{listing.title}</h2>
        <div>
          <h2 className={styles.price}>{listing.price_per_day} Р</h2>
          <p>за сутки</p>
        </div>
      </div>
      <p className={styles.desc}>{listing.description}</p>
      <div className={styles.descUnder}>
        <div className={styles.descUnderLeft}>
          <div className={styles.metro}>
            <svg
              fill="#664ef9"
              width="20px"
              height="20px"
              viewBox="0 0 24 24"
              role="img"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>Moscow Metro icon</title>
              <path d="M16.603 11.85l-2.481-6.26-2.092 3.66-2.092-3.66-2.481 6.262H6.74v.941h3.736v-.941h-.553l.538-1.555 1.569 2.57 1.569-2.57.538 1.555h-.553v.941h3.751v-.941zm5.335-1.912A9.933 9.933 0 0 0 12 0C6.516 0 2.062 4.453 2.062 9.938c0 2.75 1.121 5.23 2.914 7.023a.804.804 0 0 0 1.375-.568.825.825 0 0 0-.239-.582 8.303 8.303 0 0 1-2.42-5.873c0-4.588 3.72-8.324 8.308-8.324 4.588 0 8.324 3.736 8.324 8.324a8.289 8.289 0 0 1-2.436 5.888l-7.024 7.023L12 24l7.039-7.039a9.891 9.891 0 0 0 2.899-7.023Z" />
            </svg>
            <p className={styles.metroText}>{listing.metro}</p>
          </div>

          {listing.district && (
            <div className={styles.metro}>
              <svg
                xmlns="http://www.w3.org/2000/svg"
                fill="#664ef9"
                width="20px"
                height="20px"
                viewBox="0 0 24 24"
              >
                <path d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z" />
              </svg>
              <p className={styles.metroText}>{listing?.district}</p>
            </div>
          )}
        </div>
        <button className={styles.fav}>
          <FaHeart />В избранное
        </button>
      </div>
    </div>
  );
};
