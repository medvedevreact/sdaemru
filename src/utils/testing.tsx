import { MemoryRouter, MemoryRouterProps } from "react-router-dom";
import { render, RenderOptions } from "@testing-library/react";
import { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { store } from "../store";

interface RenderWithRouterOptions
  extends MemoryRouterProps,
    Omit<RenderOptions, "wrapper"> {}

export const renderWithRouter = (
  component: ReactElement,
  options: RenderWithRouterOptions = {}
) => {
  const Wrapper = ({ children }: { children?: ReactNode }) => (
    <MemoryRouter {...options}>
      <Provider store={store}>{children}</Provider>
    </MemoryRouter>
  );

  return render(component, { wrapper: Wrapper, ...options });
};
