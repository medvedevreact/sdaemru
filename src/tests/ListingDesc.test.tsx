import "@testing-library/jest-dom";
import { renderWithRouter } from "../utils/testing";
import { ListingDesc } from "../components/ListingDesc/ListingDesc";
import { AutoItem } from "../types";
import { screen } from "@testing-library/react";

const mockItem: AutoItem = {
  id: "1",
  title: "Toyota Prius",
  description: "Экономичный гибрид",
  price_per_day: 1500,
  location: "Москва",
  metro: "Таганская",
  district: "Центральный",
  type: "Седан",
  photo: ["car.jpg"],
  owner: {
    name: "Иван Иванов",
    phone: "1234567890",
    email: "ivan@example.com",
  },
};

describe("desc component", () => {
  test("render", () => {
    renderWithRouter(<ListingDesc listing={mockItem} />);
    expect(screen.getByRole("desc")).toMatchSnapshot();
  });
});
