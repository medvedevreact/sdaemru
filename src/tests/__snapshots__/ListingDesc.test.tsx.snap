// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`desc component render 1`] = `
<div
  class="descUpper"
  role="desc"
>
  <h2>
    Toyota Prius
  </h2>
  <div>
    <h2
      class="price"
    >
      1500
       Р
    </h2>
    <p>
      за сутки
    </p>
  </div>
</div>
`;
